freqtrade/exchange/exchange.py

```diff
@@ -1717,6 +1741,161 @@ def set_margin_mode(self, pair: str, collateral: Collateral, params: dict = {}):
         except ccxt.BaseError as e:
             raise OperationalException(e) from e
 
+    @retrier
+    def _get_mark_price_history(self, pair: str, since: int) -> Dict:
+        """
+        Get's the mark price history for a pair
+        :param pair: The quote/base pair of the trade
+        :param since: The earliest time to start downloading candles, in ms.
+        """
+
+        try:
+            candles = self._api.fetch_ohlcv(
+                pair,
+                timeframe="1h",
+                since=since,
+                params={
+                    'price': self._ft_has["mark_ohlcv_price"]
+                }
+            )
+            history = {}
+            for candle in candles:
+                d = datetime.fromtimestamp(int(candle[0] / 1000), timezone.utc)
+                # Round down to the nearest hour, in case of a delayed timestamp
+                # The millisecond timestamps can be delayed ~20ms
+                time = datetime(
+                    d.year,
+                    d.month,
+                    d.day,
+                    d.hour,
+                    tzinfo=timezone.utc
+                ).timestamp() * 1000
```

Comment:
we do provide a method for this  in `timeframe_to_prev_date` ... 
while it sounds like the wrong approach as we're not dealing with timeframes, by using `1h` as timeframe, you can get this quite nicely ... ;)